srcpkgs/webkit2gtk/template

```diff
@@ -132,21 +134,132 @@ pre_configure() {
 	esac
 }
 
+post_configure() {
+	mkdir -p webkit2gtk-41
+	pushd webkit2gtk-41
+		configure_args=${configure_args//-DUSE_SOUP2=ON/}
+		configure_args=${configure_args//-DENABLE_WEBDRIVER=OFF/}
+		do_configure
+	popd
+	mkdir -p webkit2gtk-50
+	pushd webkit2gtk-50
+		configure_args+=" -DUSE_GTK4=ON -DENABLE_WEBDRIVER=OFF"
+		do_configure
+	popd
+}
+
+pre_build() {
+	export NINJA_STATUS="[1/3][%f/%t] "
+}
+
+post_build() {
+	export NINJA_STATUS="[2/3][%f/%t] "
+	pushd webkit2gtk-41
+		do_build
+	popd
+	export NINJA_STATUS="[3/3][%f/%t] "
+	pushd webkit2gtk-50
+		do_build
+	popd
+}
+
 post_install() {
+	pushd webkit2gtk-41
+		do_install
+	popd
+	pushd webkit2gtk-50
+		do_install
+	popd
+
 	vlicense Source/WebCore/LICENSE-APPLE
 	vlicense Source/WebCore/LICENSE-LGPL-2.1
 	vlicense Source/WebCore/LICENSE-LGPL-2
 }
 
 webkit2gtk-devel_package() {
+	build_style=meta
+	depends="webkit2gtk40-devel ${sourcepkg}>=${version}_${revision}"
+	short_desc+=" - development files (transitional package)"
+}
+
+webkit2gtk40_package() {
+	depends="${_all_depends}"
+	short_desc="GTK+3 port of the WebKit2 browser engine (soup2)"
+	pkg_install() {
+		vmove usr/lib/webkit2gtk-4.0
+		vmove usr/libexec/webkit2gtk-4.0
+		if [ "$build_option_gir" ]; then
+			vmove "usr/lib/girepository-1.0/*-4.0.typelib"
+		fi
+		vmove "usr/lib/*-4.0.so.*"
+	}
+}
+
+webkit2gtk40-devel_package() {
 	depends="gtk+3-devel libsoup-devel ${sourcepkg}>=${version}_${revision}"
-	short_desc+=" - development files"
+	short_desc="GTK+3 port of the WebKit2 browser engine (soup2) - development files"
+	pkg_install() {
+		vmove usr/include/webkitgtk-4.0
+		vmove "usr/lib/pkgconfig/*-4.0.pc"
+		if [ "$build_option_gir" ]; then
+			vmove "usr/share/gir-1.0/*-4.0.gir"
+			vmove "usr/share/gtk-doc/html/*-4.0"
+		fi
+		vmove "usr/lib/*-4.0.so"
+	}
+}
+
+webkit2gtk41_package() {
+	depends="${_all_depends}"
+	short_desc="GTK+3 port of the WebKit2 browser engine (soup3)"
+	pkg_install() {
+		vmove usr/bin/WebKitWebDriver
+		vmove usr/libexec/webkit2gtk-4.1
+		vmove usr/lib/webkit2gtk-4.1
+		if [ "$build_option_gir" ]; then
+			vmove "usr/lib/girepository-1.0/*-4.1.typelib"
+		fi
+		vmove "usr/lib/*-4.1.so.*"
+	}
+}
```

Comment:
This should just be in the standard `webkit2gtk` package. (I choose this one both because I modified the default configure args to produce, I believe, v4.1; and also because this seems to be the only subpackage that ships an executable in `usr/bin/WebKitWebDriver`. If you think one of the other two should be the "preferred" or "default" version, leave that one in the main `webkit2gtk` package.)

Reply:
webkit2gtk40 is depended on by the webkit2gtk package in order to avoid having to rebuild all packages depending on webkit2gtk-4.0.
Currently the webkit2gtk package contains common files depended on by all subpackages (i.e. the locale files), although it might be better to have a webkit2gtk-common?
Imo, it's better to leave them separated, then we can decide which one is 

> also because this seems to be the only subpackage that ships an executable in `usr/bin/WebKitWebDriver`

That executable is meant to be used by say, C# or python applications for accessing the contents of webpages. (i.e. for testing purposes)
https://developer.mozilla.org/en-US/docs/Web/WebDriver

Similar to geckodriver (they both implement the same api).

Reply:
In light of that, the common stuff should definitely be moved into a common subpackage that each `lib*` package can depend on. The `webkit2gtk` package itself should could either contain whatever the "default" version is (if a sensible default can be defined).

Reply:
Is it okay to be a metapackage pointing to the default package?